Sources/LanguageServerProtocol/Requests/CallHierarchyIncomingCallsRequest.swift

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// The request is sent from the client to the server to resolve incoming calls for a given call hierarchy item.
+/// It is only issued if a server registers for the `textDocument/prepareCallHierarchy` request.
+public struct CallHierarchyIncomingCallsRequest: RequestType {
+  public static let method: String = "callHierarchy/incomingCalls"
+  public typealias Response = [CallHierarchyIncomingCall]?
+
+  public var item: CallHierarchyItem
+
+  public init(item: CallHierarchyItem) {
+    self.item = item
+  }
+}
+
+/// Represents a caller (an incoming call) - an item that makes a call of the original `item`.
+public struct CallHierarchyIncomingCall: ResponseType, Hashable {
+  /// The item that makes the call.
+  public var from: CallHierarchyItem
+
+  /// The ranges at which the calls appear. This is relative to the caller
+  /// denoted by `from`.
+  public var fromRanges: [Range<Position>]
```

Comment:
I haven't looked into this much - is this saying that these ranges are relative to `from.range`?  That's kind of weird.  Also, is it just relative to the start position or does it need to be contained within that range like selectionRange does?

Also, this probably needs to have a similar encoding trick played as we have for `@CustomCodable<PositionRange>`.

Reply:
Reworded and swapped to the customcodable